subsys/bluetooth/scan.c

```diff
@@ -632,6 +648,10 @@ static bool adv_uuid_compare(const struct bt_data *data, u8_t uuid_type)
 			      &uuid_filter->uuid[i])) {
 			uuid_match_cnt++;
 
+			control->filter_status.uuid.uuid[i] =
+				uuid_filter->uuid[i].uuid;
+			control->filter_status.uuid.count++;
```

Comment:
Using two counters, maybe just assign uuid.count = uuid_match_cnt at the end?

Reply:
Ok, Changed